File: frontend/src/types/repository.ts
export interface RepositoryValidationRequest {
  url: string;
  token: string;
  remember_token: boolean;
}

export interface RepositoryValidationResponse {
  valid: boolean;
  owner: string;
  repo: string;
  default_branch: string;
  private: boolean;
  message: string;
}

File: backend/app/schemas/repository.py
from pydantic import BaseModel, Field
from typing import Optional


class RepositoryValidateRequest(BaseModel):
    url: str = Field(..., examples=["https://github.com/arshad-rahman/releasepilot"])
    token: str = Field(..., min_length=10)
    remember_token: bool = False


class RepositoryValidateResponse(BaseModel):
    valid: bool
    owner: Optional[str] = None
    repo: Optional[str] = None
    default_branch: Optional[str] = None
    private: Optional[bool] = None
    message: str
